vision/interface.py

```diff
@@ -1,7 +1,7 @@
 
 """The Environment class will manage the other objects in the environment"""
```

Comment:
This looks good but is definitely needs a unit test, ensuring no crashes and no data loss. Have 2-3 agents trying to update at the same time and maybe 1 agent trying to read from.